receiver/kafkametricsreceiver/broker_scraper.go

```diff
@@ -51,11 +58,46 @@ func (s *brokerScraper) scrape(context.Context) (pmetric.Metrics, error) {
 		s.client = client
 	}
 
+	now := pcommon.NewTimestampFromTime(time.Now())
+	rb := s.mb.NewResourceBuilder()
+	rb.SetClusterAlias(s.config.ClusterAlias)
+
 	brokers := s.client.Brokers()
+	s.mb.RecordKafkaBrokersDataPoint((now), int64(len(brokers)))
 
-	s.mb.RecordKafkaBrokersDataPoint(pcommon.NewTimestampFromTime(time.Now()), int64(len(brokers)))
+	if s.clusterAdmin == nil {
```

Comment:
Question out of ignorance, since `clusterAdmin` is a value variable, not a pointer, can it ever be nil?

Reply:
Yes It can, clusterAdmin is sarama.ClusterAdmin interface, It will be nil if it's not initialised.